src/open_samus_returns_rando/files/schema.json

```diff
@@ -255,20 +255,11 @@
             },
             "default": {}
         },
-        "suit_reductions": {
-            "description": "Set the damage reduction per suit.",
-            "type": "object",
-            "properties": {
-                "varia_suit_reduction": {
-                    "type": "number",
-                    "default": 0.75
-                },
-                "gravity_suit_reduction": {
-                    "type": "number",
-                    "default": 0.5
-                }
-            },
-            "default": {}
+        "required_dna": {
+            "description": "The amount of Metroid DNA required to access Proteus Ridley.",
+            "type": "integer",
+            "minimum": 0,
+            "default": 39
```

Comment:
Just a comment that this breaks all previous inputs / patcher.json files such that they are not beatable anymore without manual modifications (unless they have all 39 DNA placed).
Don't know if you want to account for that in the next version number.

Reply:
Then you give free access to ridley because this will be always true:
https://github.com/randovania/open-samus-returns-rando/blob/f6e76e0251aca6a33cf3d8a0798a23de82a781ef/src/open_samus_returns_rando/files/templates/custom_init.lua#L76-L81

I can't think of any way besides writing a migration like: "If required_dna is not present, count the amount of DNA in starting inventory and set required_dna to 39 - DNA in inventory".
Or just break old files and bump major version.
I'm wondering if anyone is directly using patcher files besides the few people following the development.

Reply:
> I'm wondering if anyone is directly using patcher files besides the few people following the development.

AFAIK RDV is currently the only downstream. So for least effort i'd just bump major+break compat

Reply:
Feels wrong to bump major for something like this. I'd rather restructure the schema if we were to bump major.

Reply:
There's nothing wrong about it and it shouldn't feel wrong either. A new feature that isnt (in a simple way) backwards compatible is a perfectly valid reason for a major bump.

Reply:
> > I'm wondering if anyone is directly using patcher files besides the few people following the development.
> 
> AFAIK RDV is currently the only downstream. So for least effort i'd just bump major+break compat

It's not about RDV. RDV takes a rdvgame and creates a patcher file. Of course it should create the appropriate file once this is merged into RDV even for older rdvgame files. So, RDV already does a migration and makes sure that old rdvgame files do not break.
It's about the fact that with the default you use in the schema, the programm still executes but the output is different such that the game is not beatable anymore, which, at least to me, is a breaking change.
Or you use 0 as default but then the game also changes behaviour because you can directly access ridley.

I would just bump the major version but also it's very likely that nobody will care because there is no one using the patcher directly.

Reply:
> nobody will care because there is no one using the patcher directly.

That's what I meant. The (currently known) only user of OSRR is RDV ,and RDV has migrations in place to not be affected by this.

Reply:
> I can't think of any way besides writing a migration like: "If required_dna is not present, count the amount of DNA in starting inventory and set required_dna to 39 - DNA in inventory".

Honestly, this doesn't sound like a bad idea. Should be easy to add.